Modifier son profil

TuxBoy opened the pull request:
Add User case classes and unit tests

- [X] Domain
- [X] unit tests
- [X] implémentation du domain dans le UserInterface
- [X] Test d'intégration

TuxBoy committed:
feature(use-case): Add Update profile user case

Add User case classes and unit tests

TuxBoy committed:
feature(use-case): Add upload avatar on participant

Add avatar, Uploader provider and unit tests

TuxBoy committed:
feature(UserInterface): add update profile controller

add controller, implements Presenter & unit and integration test

TBoileau commented on domain/src/Security/Exception/NonValidAvatarException.php:
```diff
@@ -0,0 +1,14 @@
+ *
+ * @package TBoileau\CodeChallenge\Domain\Security\Exception
+ */
+class NonValidAvatarException extends InvalidArgumentException
```
`NonValid` ou `Invalid`

TBoileau commented on domain/src/Security/Presenter/UpdateProfilePresenterInterface.php:
```diff
@@ -0,0 +1,11 @@
+
+interface UpdateProfilePresenterInterface
+{
+
```
blank line inutile

TBoileau commented on domain/src/Security/Provider/UploaderProviderInterface.php:
```diff
@@ -0,0 +1,19 @@
+
+interface UploaderProviderInterface
+{
+
```
blank line inutile

TBoileau commented on domain/src/Security/Provider/UploaderProviderInterface.php:
```diff
@@ -0,0 +1,19 @@
+     * @param Uploader $uploader
+     * @return string
+     */
+    public function upload(Uploader $uploader): string;
```
Essaie de travailler  avec des interfaces, inversion de dépendance : on dépend d'une abstraction et non d'une implémentation

TBoileau commented on domain/src/Security/Request/UpdateProfileRequest.php:
```diff
@@ -0,0 +1,84 @@
+
+    private ?Uploader $avatarPath = null;
+
+    public function __construct(UuidInterface $id, string $email, string $pseudo, ?Uploader $avatarPath = null)
```
Même remarque plus haut, on travaille avec des interface pour `Uploader`, est-ce que le wording te semble bon ? Pour Uploader est un service d'upload

TBoileau commented on src/UserInterface/DataTransferObject/Participant.php:
```diff
@@ -0,0 +1,108 @@
+    private ?string $pseudo = null;
+
+    /**
+     * @NonValidExtension()
```
Tu testes l'extension alors qu'il faudrait plutôt tester le mime type

TuxBoy committed:
feature(uploader): Use an interface to uploader & validation image with mime type

Adding UploaderInterface to stop used implements class & add validation image by mime type.

TuxBoy committed:
fix(UseCase): bad parameter sent to image validation

In UpdateProfileRequest validate method, sent image for validation without originalName param
